### PR TITLE
Update startup script

### PR DESCRIPTION
**Pull request by kikomanaskov333**
Updated the startup script to fix the foia analythical reports showing no data.

### CODE REVIEW

File: Packer/ArkCase-FOIA/startup.sh [hunk 1]
```diff
@@ -43,6 +44,10 @@ if [ ! -e /var/lib/admin-password-changed ]; then
     # Update the password in the portal configuration
     sed -i "19s/@rKc@3e/$clear_password/g" "${rootdir}/data/arkcase-home/.arkcase/acm/acm-config-server-repo/arkcase-portal-server.yaml"
 
+    # Update the password in the foia analytical reports configuration
+    sed -i "s/ARKCASE_PASS=.*/ARKCASE_PASS=$clear_password/g" "${rootdir}/install/pentaho/foia-reports-dw-2021.01/config/arkcase_config.properties"
```
Comment (bojmil-armedia) [inline Packer/ArkCase-FOIA/startup.sh]: What is `2021.01` ? Should it be the version variable?

File: Packer/ArkCase-FOIA/startup.sh [hunk 2]
```diff
@@ -114,3 +119,31 @@ systemctl start alfresco
...
 systemctl start arkcase
 systemctl start haproxy
+
+## Wait for ArkCase to fully start
+timeout_min=60  # Timeout: 1h
+timer_min=0
+echo "Wait for ArkCase to fully start..."
+while true; do
+    sleep 60  # Wait for 1'
+    if sudo grep 'org.apache.catalina.startup.Catalina.start Server startup in \[.*\] milliseconds' ${rootdir}/log/arkcase/catalina.out > /dev/null 2>&1; then
```
Comment (bojmil-armedia) [inline Packer/ArkCase-FOIA/startup.sh]: Just to let you know that some versions of Tomcat have different message. On the current one this will work.

Comment (bojmil-armedia) [reply Packer/ArkCase-FOIA/startup.sh]: Maybe trying if the port is open is a better test that Arkcase has started.

Comment (david-oc-miller) [reply Packer/ArkCase-FOIA/startup.sh]: The port opens long before the ArkCase webapp is responsive ... we could check if the /arkcase URL responses with a 200 or 302, versus a 404 or 500.

File: Packer/ArkCase-FOIA/startup.sh [hunk 3]
```diff
@@ -114,3 +119,31 @@ systemctl start alfresco
...
+            echo "ERROR: ArkCase didn't start within $timeout_min minutes"
+            exit 1
+        fi
+        echo -n .
+    fi
+done
+echo
+echo "ArkCase fully started"
+
+# Run kitchen scipt for foia analytical reports
```
Comment (fabricetriboix) [inline Packer/ArkCase-FOIA/startup.sh]: Typo: `scipt` -> `script`

File: Packer/ArkCase-FOIA/startup.sh (hunk 1 above)
Comment (kikomanaskov333) [reply Packer/ArkCase-FOIA/startup.sh]: Yes, I have defined it in the script but somehow I forgot to use it here. Now it is ok :)

File: Packer/ArkCase-FOIA/startup.sh (hunk 2 above)
Comment (kikomanaskov333) [reply Packer/ArkCase-FOIA/startup.sh]: Now we're waiting until arkcase URL responses with a 200

File: Packer/ArkCase-FOIA/startup.sh (hunk 3 above)
Comment (kikomanaskov333) [reply Packer/ArkCase-FOIA/startup.sh]: Tnx, @fabricetriboix

File: Packer/ArkCase-FOIA/startup.sh [hunk 4]
```diff
@@ -114,3 +119,14 @@ systemctl start alfresco
 systemctl start config-server
 systemctl start arkcase
 systemctl start haproxy
+
+## Wait for ArkCase to become available on /arkcase
+timeout 600 bash -c 'while [[ "$(curl --insecure -s -o /dev/null -w ''%{http_code}'' https://localhost/arkcase/login)" != "200" ]]; do sleep 5; done'
```
Comment (bojmil-armedia) [inline Packer/ArkCase-FOIA/startup.sh:124]: Is 600 seconds enough on first start? Or this is not the first start?

**Review by bojmil-armedia — DISMISSED** [review]

File: Packer/ArkCase-FOIA/startup.sh (hunk 4 above)
Comment (kikomanaskov333) [reply Packer/ArkCase-FOIA/startup.sh:124]: It is not a first start. The first is in the packer process where AMI is built

**Review by fabricetriboix — DISMISSED** [review]

**Review by bojmil-armedia — APPROVED** [review]

**Pull request merged**